### DISCUSSION TITLE
New registration cannot be logged

### DISCUSSION THREAD

**Discussion by yangshengjie4u** [spatie/laravel-activitylog#925] [Q&A] [answered]
I'm using v3.17 and here is my code in User model
```
......
protected static $logFillable = true;
protected static $logOnlyDirty = true;

protected static $logName = 'User';
protected static $recordEvents = ['created','updated','deleted'];

public function tapActivity(Activity $activity, string $eventName)
{
    // collect some info to log
}
......
```
1. New registered user info is saved at `properties` field, but the `causer_type` and `causer_id` are both null. How to set `causer_id` as the new user's ID and put User model into `causer_type`?
2. How to change the `log_name`, 'default', if I want to take a log in Controller?

**Comment by Gummibeer** [comment, marked as answer]
1. You could use the `tapActivity()` method and the passed-in event name to set the `causer` to `$this` if the event is `created` and the `causer` is empty
2. pass your wanted log name as the first argument to the `activity('my_log_name')` function
